$MirOS: ports/emulators/qemu/patches/patch-fpu_softfloat-native_c,v 1.2 2006/12/21 22:59:30 tg Exp $
--- fpu/softfloat-native.c.orig	Sun Sep  4 17:11:09 2005
+++ fpu/softfloat-native.c	Thu Dec 21 21:42:21 2006
@@ -1,8 +1,13 @@
 /* Native implementation of soft float functions. Only a single status
    context is supported */
 #include "softfloat.h"
+#include "gnu-c99-math.h"
 #include <math.h>
 
+#ifdef _BSD
+#include <sys/param.h>
+#endif
+
 void set_float_rounding_mode(int val STATUS_PARAM)
 {
     STATUS(float_rounding_mode) = val;
@@ -22,7 +27,7 @@ void set_floatx80_rounding_precision(int
 }
 #endif
 
-#if defined(_BSD)
+#if defined(_BSD) && (!defined(MirBSD) || (MirBSD < 0x09A6))
 #define lrint(d)		((int32_t)rint(d))
 #define llrint(d)		((int64_t)rint(d))
 #endif
